runtime/android/core/src/org/xwalk/core/XWalkLibraryLoader.java

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.xwalk.core;
+
+import android.app.DownloadManager;
+import android.app.DownloadManager.Request;
+import android.app.DownloadManager.Query;
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.os.Environment;
+import android.util.Log;
+
+import java.lang.Thread;
+
+import junit.framework.Assert;
+
+import org.xwalk.core.XWalkLibraryInterface.DecompressionListener;
+import org.xwalk.core.XWalkLibraryInterface.DownloadListener;
+import org.xwalk.core.XWalkLibraryInterface.InitializationListener;
+
+/*
+ * <p>Steps for loading the library:
+ * prepareToInit() -> startDecompression() -> initXWalkLibrary() ->
+ * [if not matched -> install library -> initXWalkLibrary()] ->
+ * startInitialization() -> over</p>
+*/
+public class XWalkLibraryLoader {
+    private static final String XWALK_APK_NAME = "XWalkRuntimeLib.apk";
+    private static final String TAG = "XWalkLib";
+
+    private static AsyncTask<Void, Integer, Integer> sActiveTask;
+    private static boolean sPreparedToInit = false;
+    private static boolean sCheckedDecompression = false;
+
+    public static void prepareToInit() {
+        sPreparedToInit = true;
+        XWalkCoreWrapper.handlePreInit();
+    }
```

Comment:
could remove prepareToInit, remove sPreparedToInit, and init XWalkCoreWrapper.sReservedObjects in its definition, as:
private static LinkedList<Object> sReservedObjects = new LinkedList<Object>();

Then check for sPreparedToInit in line 45 could remove also.